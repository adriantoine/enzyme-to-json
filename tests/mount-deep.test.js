/* eslint-env jest */

import React from 'react';
import Enzyme, {mount} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import mountToJson from '../src/mount';
import {
  BasicPure,
  BasicWithUndefined,
  BasicWithAList,
  ComponentWithAZeroChildren,
  ArrayRender,
} from './fixtures/pure-function';
import {
  BasicClass,
  ClassWithPure,
  ClassWithDirectPure,
  ClassWithDirectComponent,
  ClassWithNull,
  ClassArrayRender,
} from './fixtures/class';

Enzyme.configure({adapter: new Adapter()});
const deepOptions = {mode: 'deep'};

it('converts basic pure mount', () => {
  const mounted = mount(
    <BasicPure className="pure">
      <strong>Hello!</strong>
    </BasicPure>,
  );

  expect(mountToJson(mounted, deepOptions)).toMatchSnapshot();
});

it('converts pure mount with mixed children', () => {
  const mounted = mount(<BasicPure>Hello {'world'}!</BasicPure>);

  expect(mountToJson(mounted, deepOptions)).toMatchSnapshot();
});

it('converts basic class mount', () => {
  const mounted = mount(
    <BasicClass className="class">
      <strong>Hello!</strong>
    </BasicClass>,
  );

  expect(mountToJson(mounted, deepOptions)).toMatchSnapshot();
});

it('converts class mount with mixed children', () => {
  const mounted = mount(<BasicClass>Hello {'world'}!</BasicClass>);

  expect(mountToJson(mounted, deepOptions)).toMatchSnapshot();
});

it('converts a class mount with a pure function in it', () => {
  const mounted = mount(
    <ClassWithPure className="class">
      <strong>Hello!</strong>
    </ClassWithPure>,
  );

  expect(mountToJson(mounted, deepOptions)).toMatchSnapshot();
});

it('converts a class mount with a pure function in it as a direct child', () => {
  const mounted = mount(
    <ClassWithDirectPure className="class">
      <strong>Hello!</strong>
    </ClassWithDirectPure>,
  );

  expect(mountToJson(mounted, deepOptions)).toMatchSnapshot();
});

it('converts a class mount with a class component in it as a direct child', () => {
  const mounted = mount(
    <ClassWithDirectComponent className="class">
      <strong>Hello!</strong>
    </ClassWithDirectComponent>,
  );

  expect(mountToJson(mounted, deepOptions)).toMatchSnapshot();
});

it('converts function components with render returning top level arrays', () => {
  const mounted = mount(
    <ArrayRender>
      <strong>Hello!</strong>
    </ArrayRender>,
  );

  expect(mountToJson(mounted, deepOptions)).toMatchSnapshot();
});

it('converts class components with render returning top level arrays', () => {
  const mounted = mount(
    <ClassArrayRender>
      <strong>Hello!</strong>
    </ClassArrayRender>,
  );

  expect(mountToJson(mounted, deepOptions)).toMatchSnapshot();
});

it('handles a component which returns null', () => {
  const mounted = mount(<ClassWithNull />);

  expect(mountToJson(mounted, deepOptions)).toMatchSnapshot();
});

it('includes undefined props', () => {
  const mounted = mount(<BasicWithUndefined>Hello!</BasicWithUndefined>);

  expect(mountToJson(mounted, deepOptions)).toMatchSnapshot();
});

it('renders zero-children', () => {
  const mounted = mount(<ComponentWithAZeroChildren />);

  expect(mountToJson(mounted, deepOptions)).toMatchSnapshot();
});

it('renders multiple elements as a result of find', () => {
  const mounted = mount(<BasicWithAList />);

  expect(mountToJson(mounted.find('li'), deepOptions)).toMatchSnapshot();
});
