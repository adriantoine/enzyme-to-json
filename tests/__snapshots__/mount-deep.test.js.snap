// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`converts a class mount with a class component in it as a direct child 1`] = `
<div
  className="class-with-pure nested-pure"
  onClick={[Function]}
>
  <div
    className="basic-pure nested-pure"
    onClick={[Function]}
  >
    <div
      className="group"
      id="group-id"
    >
      <span>
        <span>
          <span>
            <strong>
              Hello!
            </strong>
          </span>
          <span
            className="empty"
          />
        </span>
        <span
          className="empty"
        />
      </span>
      <span
        className="empty"
      />
    </div>
  </div>
</div>
`;

exports[`converts a class mount with a pure function in it 1`] = `
<div
  className="class-with-pure class"
  onClick={[Function]}
>
  <div
    className="basic-pure nested-pure"
    onClick={[Function]}
  >
    <div
      className="group"
      id="group-id"
    >
      <span>
        <span>
          <strong>
            Hello!
          </strong>
        </span>
        <span
          className="empty"
        />
      </span>
      <span
        className="empty"
      />
    </div>
  </div>
</div>
`;

exports[`converts a class mount with a pure function in it as a direct child 1`] = `
<div
  className="basic-pure nested-pure"
  onClick={[Function]}
>
  <div
    className="group"
    id="group-id"
  >
    <span>
      <span>
        <strong>
          Hello!
        </strong>
      </span>
      <span
        className="empty"
      />
    </span>
    <span
      className="empty"
    />
  </div>
</div>
`;

exports[`converts basic class mount 1`] = `
<div
  className="basic-class class"
  onClick={[Function]}
>
  <div
    className="group"
    id="group-id"
  >
    <span>
      <strong>
        Hello!
      </strong>
    </span>
    <span
      className="empty"
    />
  </div>
</div>
`;

exports[`converts basic pure mount 1`] = `
<div
  className="basic-pure pure"
  onClick={[Function]}
>
  <div
    className="group"
    id="group-id"
  >
    <span>
      <strong>
        Hello!
      </strong>
    </span>
    <span
      className="empty"
    />
  </div>
</div>
`;

exports[`converts class components with render returning top level arrays 1`] = `
Array [
  <div
    className="test"
    key="test"
  >
    Test
  </div>,
  <div
    className="test2"
    key="test2"
  >
    Test 2
  </div>,
  <div
    className="child"
    key="child"
  >
    <strong>
      Hello!
    </strong>
  </div>,
]
`;

exports[`converts class mount with mixed children 1`] = `
<div
  className="basic-class undefined"
  onClick={[Function]}
>
  <div
    className="group"
    id="group-id"
  >
    <span>
      Hello 
      world
      !
    </span>
    <span
      className="empty"
    />
  </div>
</div>
`;

exports[`converts function components with render returning top level arrays 1`] = `
Array [
  <div
    className="test"
    key="test"
  >
    Test
  </div>,
  <div
    className="test2"
    key="test2"
  >
    Test 2
  </div>,
  <div
    className="child"
    key="child"
  >
    <strong>
      Hello!
    </strong>
  </div>,
]
`;

exports[`converts pure mount with mixed children 1`] = `
<div
  className="basic-pure undefined"
  onClick={[Function]}
>
  <div
    className="group"
    id="group-id"
  >
    <span>
      Hello 
      world
      !
    </span>
    <span
      className="empty"
    />
  </div>
</div>
`;

exports[`handles a component which returns null 1`] = `""`;

exports[`includes undefined props 1`] = `
<button>
  Hello
</button>
`;

exports[`renders multiple elements as a result of find 1`] = `
Array [
  <li>
    0
  </li>,
  <li>
    1
  </li>,
]
`;

exports[`renders zero-children 1`] = `
<div>
  0
</div>
`;
